Add support for uploading new version from IO

danthompson opened the pull request:
This commit adds support for uploading a new file version via IO instead of requiring a `file_path`.

```
upload_new_version_of_file_from_io(...)
```

I've modeled this after the `upload_file` and `upload_file_from_io` methods so the non-IO variant utilizes the new method. Of note, when adding tests I also needed to bump the version count being asserted on.

xhocquet commented:
Thanks for the contribution @danthompson ! At first glance this seems great and conforms to existing patterns. A couple things I would ask - 
 * Bump the minor version since this is a 'new' feature
 * Add this call to the README next to the existing path method to let people know this is available

Cheers! 🙏🏻

danthompson commented:
Thanks for the quick response @xhocquet.

I've added the method to the README, and bumped the minor version as requested. The `upload_file_from_io` was also missing from the README so I've added that as well.

xhocquet commented:
Tested and everything seems to be working as expected with no regressions. Thanks you!
